src/main/proto/th2_grpc_common/common.proto

```diff
@@ -200,8 +200,18 @@ message Checkpoint {
   string id = 1;
   map<string, DirectionCheckpoint> session_alias_to_direction_checkpoint = 2;
 
+  message CheckpointData {
+    int64 sequence = 1; // Message sequence number in session
+    google.protobuf.Timestamp timestamp = 2; // Message creation timestamp
+  }
+
   message DirectionCheckpoint {
-    map<int32, int64> direction_to_sequence = 1;
+    /*
+    * Values are not considered in check1 since the 2.10.0 version. Please use direction_to_checkpoint_data instead.
+    * This field will be removed in future major releases.
```

Comment:
I think we still will need to use this map until we remove it completely. So, we cannot say that check1 will not consider that field at all. In check1 we will need to check if one of the fields is set and convert the old one if the old one was used. But if both fields are set we should report an error